File: Scripts/generated_codes/java/268.java
/*

*/

import java.util.*;
import java.io.*;
import java.math.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        for (int i=0; i<t; i++) {
            String str = br.readLine();
            String[] strArr = str.split("\\s+");
            int DS = Integer.parseInt(strArr[0]);
            int DT = Integer.parseInt(strArr[1]);
            int D = Integer.parseInt(strArr[2]);
            double ds = Math.pow(DS,2);
            double dt = Math.pow(DT,2);
            double d = Math.pow(D,2);
            double result = 0;
            if(D<DS+DT){
                result = Math.sqrt(ds+dt-d)/2;
            }
            else if(D==DS+DT){
                result = 0;
            }
            else{
                result = Math.sqrt(d-ds-dt)/2;
            }
            System.out.println(result);            
        }
    }
}

File: Scripts/generated_codes/python/codechef_responses/268.py
*/
#include <stdio.h>
#include <math.h>
#include <stdlib.h>
int main()
{
	int t,i,j;
	scanf("%d",&t);
	float a[t],b[t],c[t];
	for(i=0;i<t;i++)
	{
		scanf("%f%f%f",&a[i],&b[i],&c[i]);
	}
	for(j=0;j<t;j++)
	{
		printf("%.3f\n",sqrt(c[j]*c[j]-(a[j]-b[j])*(a[j]-b[j])));
	}
	return 0;
}